### PR TITLE
add empty() to PriorityConnPoolMap

### PR DESCRIPTION
**Pull request by lambdai**
Commit Message:
Speed up a little bit since pool is very unlikely empty().

Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by davinci26 — COMMENTED** [review]
LGTM thanks, just a nit suggestion, do you want to add a comment in the function `size()` to say that it iterates over the map so that callers know to prefer `empty()`

**Comment by lambdai** [comment]
> LGTM thanks, just a nit suggestion, do you want to add a comment in the function `size()` to say that it iterates over the map so that callers know to prefer `empty()`

Linter should have reported it. After I add the empty(), my vscode starts to report it. 
Also, the suggestion is applied to all stl container. Let's assume people think it as a stl container :)

**Review by davinci26 — DISMISSED** [review]
If the linter already reports it, then this is a way stronger signal than a docs comment

**Review by davinci26 — APPROVED** [review]
LGTM, onto a second @envoyproxy/envoy-maintainers to review and merge

File: source/common/upstream/conn_pool_map.h [hunk 1]
```diff
@@ -46,6 +46,11 @@ template <typename KEY_TYPE, typename POOL_TYPE> class ConnPoolMap {
    */
   size_t size() const;
 
+  /**
+   * @return true if the pools are empty.
+   */
+  size_t empty() const;
```
Comment (mergeconflict) [inline source/common/upstream/conn_pool_map.h:52]: should this return `bool` instead of `size_t`?

Comment (ggreenway) [reply source/common/upstream/conn_pool_map.h:52]: Yes, it definitely should. @lambdai can you post a followup to fix that?

Comment (lambdai) [reply source/common/upstream/conn_pool_map.h:52]: oops. Sure

**Pull request merged**